src/o365/spfx/commands/project/project-rename.ts

```diff
@@ -0,0 +1,193 @@
+const uuid = require('uuid');
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  newName: string;
+  generateNewId?: boolean;
+}
+
+class SpfxProjectRenameCommand extends BaseProjectCommand {
+  public constructor() {
+    super();
+  }
+
+  public static ERROR_NO_PROJECT_ROOT_FOLDER: number = 1;
+
+  public get name(): string {
+    return commands.PROJECT_RENAME;
+  }
+
+  public get description(): string {
+    return 'Rename SharePoint Framework project';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.newName = (!(!args.options.newName)).toString();
+    telemetryProps.generateNewId = args.options.generateNewId;
+    return telemetryProps;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-n, --newName <newName>',
+        description: 'New name for the project'
+      },
+      {
+        option: '--generateNewId',
+        description: 'Generate a new solution id for the project'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.newName) {
+        return 'Required parameter newName missing';
+      }
+
+      return true;
+    };
+  }
+
+  public action(): CommandAction {
+    const cmd: Command = this;
+    return function (this: CommandInstance, args: CommandArgs, cb: (err?: any) => void) {
+      args = (cmd as any).processArgs(args);
+      (cmd as any).initAction(args, this);
+      cmd.commandAction(this, args, cb);
+    }
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      cb(new CommandError(`Couldn't find project root folder`, SpfxProjectRenameCommand.ERROR_NO_PROJECT_ROOT_FOLDER));
+      return;
+    }
+
+    const packageJson: any = this.getProject(this.projectRootPath).packageJson;
+    const projectName: string = packageJson.name;
+
+    let newId: string;
+    if (args.options.generateNewId) {
+      newId = uuid.v4();
+      if (this.debug) {
+        cmd.log('Created new solution id');
+        cmd.log(newId);
+      }
+    }
+
+    const filePaths: string[] = [
+      path.join(this.projectRootPath, 'package.json'),
+      path.join(this.projectRootPath, '.yo-rc.json'),
+      path.join(this.projectRootPath, 'config/package-solution.json'),
+      path.join(this.projectRootPath, 'config/deploy-azure-storage.json'),
+      path.join(this.projectRootPath, 'README.md')
+    ];
+
+    const replaceFileContent = (filePath: string) => {
+      if (fs.existsSync(filePath)) {
```

Comment:
Rather than wrapping the whole block in this `if` statement, you could invert it and simplify the whole block:

```ts
if (!fs.existsSync(filePath)) {
  return;
}

// ... rest of the code
```

Reply:
Thanks. Updated now.